Jenkinsfile

```diff
@@ -0,0 +1,45 @@
+pipeline {
+    agent { label 'linux-small' }
+    options {
+            buildDiscarder(logRotator(numToKeepStr:'25'))
+            disableConcurrentBuilds()
+            timestamps()
+            skipDefaultCheckout()
+    }
+    environment {
+        MVN_OPTS = '-Xmx1024M -Xss128M -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC '
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
+    }
+    stages {
+        stage('Build') {
+            steps {
+                retry(3) {
+                    checkout scm
+                }
+                timeout(time: 20, unit: 'MINUTES') {
+                    withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${MVN_OPTS} ${LINUX_MVN_RANDOM}') {
+                        sh 'mvn clean install'
+                    }
+                }
+            }
+        }
+        /*
+          Deploy stage will only be executed for deployable branches. These include master and any patch branch matching M.m.x format (i.e. 2.10.x, 2.9.x, etc...).
```

Comment:
✏️ if you update the file might be worth clarifying here on versions. These versions are for DDF not the thirdparty project.

Reply:
Should I make it more generic, like `1.0.x`, `9.1.x`? We still use the `X.X.X` version format in this pom so it technically applies, but I see what you mean.